### ISSUE TITLE
Example script fails - bash cannot found miniconda

### ISSUE THREAD

**Issue by gloriamacia** [aws-samples/amazon-sagemaker-notebook-instance-lifecycle-config-samples#29]
I tried running the default on-start, on-create scripts, aka your example. The on-create scripts does not install miniconda successfully. Namely, I am referring to this line: 
bash "$WORKING_DIR/miniconda.sh" -b -u -p "$WORKING_DIR/miniconda" 
I kindly ask you to fix this issue. Thanks!

**Comment by sema4-eculbertson** [comment]
I removed the sudo lines at the top of the file and it worked ok.
```
#!/bin/bash

set -e

# OVERVIEW
# This script installs a custom, persistent installation of conda on the Notebook Instance's EBS volume, and ensures that these custom environments
# are available as kernels in Jupyter.
# 
# The on-create script downloads and installs a custom conda installation to the EBS volume via Miniconda. Any relevant packages can be installed here 
#   1. ipykernel is installed to   
#   2. Ensure the Notebook Instance has internet connectivity to download the Miniconda installer
#
# For another example, see https://docs.aws.amazon.com/sagemaker/latest/dg/nbi-add-external.html#nbi-isolated-environment



# Install a separate conda installation via Miniconda
WORKING_DIR=/home/ec2-user/SageMaker/custom-miniconda
mkdir -p "$WORKING_DIR"
wget https://repo.anaconda.com/miniconda/Miniconda3-4.6.14-Linux-x86_64.sh -O "$WORKING_DIR/miniconda.sh"
bash "$WORKING_DIR/miniconda.sh" -b -u -p "$WORKING_DIR/miniconda" 
rm -rf "$WORKING_DIR/miniconda.sh"


# Create a custom conda environment
source "$WORKING_DIR/miniconda/bin/activate"
KERNEL_NAME="custom_python"
PYTHON="3.6"

conda create --yes --name "$KERNEL_NAME" python="$PYTHON"
conda activate "$KERNEL_NAME"

pip install --quiet ipykernel

# Customize these lines as necessary to install the required packages
pip install --quiet nltk
```

**Comment by gloriamacia** [comment]
In my case the problem was the windows invisible new line character plus 'EOF' needed to be put in ''

**Comment by sandeepksaini** [comment]
This issue can be resolved by using :

- 'EOF' -> Putting quotes at the start will help escape the "$" inside the bash script which will result in the proper variable name used inside this script.
- \EOF 

### After modifications script looks like this:
```
#!/bin/bash

set -e

# OVERVIEW
# This script installs a custom, persistent installation of conda on the Notebook Instance's EBS volume, and ensures that these custom environments
# are available as kernels in Jupyter.
# 
# The on-create script downloads and installs a custom conda installation to the EBS volume via Miniconda. Any relevant packages can be installed here 
#   1. ipykernel is installed to   
#   2. Ensure the Notebook Instance has internet connectivity to download the Miniconda installer
#
# For another example, see https://docs.aws.amazon.com/sagemaker/latest/dg/nbi-add-external.html#nbi-isolated-environment

# Use any one of the change to make this script work
#sudo -u ec2-user -i << \EOF
sudo -u ec2-user -i << 'EOF'
unset SUDO_UID
# Install a separate conda installation via Miniconda
WORKING_DIR=/home/ec2-user/SageMaker/custom-miniconda
mkdir -p "$WORKING_DIR"
wget https://repo.anaconda.com/miniconda/Miniconda3-4.6.14-Linux-x86_64.sh -O "$WORKING_DIR/miniconda.sh"
bash "$WORKING_DIR/miniconda.sh" -b -u -p "$WORKING_DIR/miniconda" 
rm -rf "$WORKING_DIR/miniconda.sh"
# Create a custom conda environment
source "$WORKING_DIR/miniconda/bin/activate"
KERNEL_NAME="custom_python"
PYTHON="3.6"
conda create --yes --name "$KERNEL_NAME" python="$PYTHON"
conda activate "$KERNEL_NAME"
pip install --quiet ipykernel
# Customize these lines as necessary to install the required packages
conda install --yes numpy
pip install --quiet boto3
EOF
```

**Comment by athewsey** [comment]
I believe this should be resolved by merge of my PR #36, if it's approved

**Comment by dmlause** [comment]
Closing given the PRs have been merged
